### PR TITLE
Adding support for Onshape

### PR DESCRIPTION
**Pull request by ghost**
Adding support for Onshape (Future of CAD) - https://www.onshape.com/

### CODE REVIEW

File: Owin.Security.Providers/Onshape/OnshapeAuthenticationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            return app.UseOnshapeAuthentication(new OnshapeAuthenticationOptions
+            {
+                AppKey = appKey,
+                AppSecret = appSecret,
+                CallbackPath = new PathString(callbackPath)
+            });
+        }
+
+        public static IAppBuilder UseOnshapeAuthentication(this IAppBuilder app, string appKey,
```
Comment (jerriep) [inline Owin.Security.Providers/Onshape/OnshapeAuthenticationExtensions.cs]: For consistency with other providers, please stick to only 2 overloads, namely: 

```
public static IAppBuilder UseOnshapeAuthentication(this IAppBuilder app, OnshapeAuthenticationOptions options)
```

and 

```
public static IAppBuilder UseOnshapeAuthentication(this IAppBuilder app, string appKey, string appSecret)
```

File: Owin.Security.Providers/Onshape/OnshapeAuthenticationOptions.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        /// </summary>
+        public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        ///     Initializes a new <see cref="OnshapeAuthenticationOptions" />
+        /// </summary>
+        public OnshapeAuthenticationOptions()
+            : base("Onshape")
+        {
+            Caption = Constants.DefaultAuthenticationType;
```
Comment (jerriep) [inline Owin.Security.Providers/Onshape/OnshapeAuthenticationOptions.cs:91]: Please set default value for CallbackPath:

```
CallbackPath = new PathString("/signin-onshape");
```

Make sure it is all lower case

File: Owin.Security.Providers/Onshape/OnshapeAuthenticationOptions.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        /// </summary>
+        /// <value>
+        ///     The back channel timeout in milliseconds.
+        /// </value>
+        public TimeSpan BackchannelTimeout { get; set; }
+
+        /// <summary>
+        ///     The request path within the application's base path where the user-agent will be returned.
+        ///     The middleware will process this request when it arrives.
+        ///     Default value is "/signin-Onshape".
```
Comment (jerriep) [inline Owin.Security.Providers/Onshape/OnshapeAuthenticationOptions.cs]: Please fix comment to reflect lowecase value, ie. "/signin-onshape"

**Comment by jerriep** [comment]
If you would like, also please consider adding an ASP.NET 5 version to the [aspnet-contrib/AspNet.Security.OAuth.Providers](https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers) repo.

Please see the [CONTRIBUTING.md](https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/CONTRIBUTING.md) file for details on how to contribute

**Comment by ghost** [comment]
@jerriep thanks! I have fixed the issues you pointed out.

**Comment by jerriep** [comment]
@tatx Thanks :) I spend Fridays on open-source, so I will test and merge it then and publish to Nuget

**Comment by ghost** [comment]
@jerriep I was wondering if you got a chance to review this pull request again. Thanks!

**Comment by ghost** [comment]
@jerriep Thanks!!

**Comment by jerriep** [comment]
@tatx  I am very sorry for the delay. I have pushed the new package to Nuget.

It may take a few minutes to show up.... :
image

**Pull request merged**